### PR TITLE
AutoDeleteColonies 1.12

### PR DESCRIPTION
**Pull request by Asherslab**
Let's retry this xD

# Changes proposed in this pull request:
- Port autoDeleteColonies to 1.12
-
-

Review please

i was not able to test this myself, it'd be great if someone could (IntelliJ refused to play along)

### CODE REVIEW

**Comment by Asherslab** [comment]
just BTW the codacy crap isn't me xD

File: src/main/java/com/minecolonies/coremod/commands/generalcommands/CheckForAutoDeletesCommand.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+        final List<Colony> colonies = ColonyManager.getColonies();
+
+        final List<Colony> coloniesToDelete = new ArrayList<>();
+
+        for (int index = 0; colonies.size() - 1 >= index; index++)
+        {
+            final Colony colony = colonies.get(index);
+
+            if (colony.canBeAutoDeleted() && Configurations.gameplay.autoDeleteColoniesInHours != 0 && colony.getLastContactInHours() >= Configurations.gameplay.autoDeleteColoniesInHours)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/commands/generalcommands/CheckForAutoDeletesCommand.java]: MAJOR Split this 187 characters long line (which is greater than 180 authorized). [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00103)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 69 issues
* ![MAJOR][MAJOR] 25 major
* ![MINOR][MINOR] 41 minor
* ![INFO][INFO] 3 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/7979536f26cf9f9f3c5c622d68a3dc6b30140ae3/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.289 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L51](https://github.com/Minecolonies/minecolonies/blob/7979536f26cf9f9f3c5c622d68a3dc6b30140ae3/src/main/java/com/minecolonies/coremod/colony/Colony.java#L51): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 46 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L51](https://github.com/Minecolonies/minecolonies/blob/7979536f26cf9f9f3c5c622d68a3dc6b30140ae3/src/main/java/com/minecolonies/coremod/colony/Colony.java#L51): class "Colony" has 91 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L216](https://github.com/Minecolonies/minecolonies/blob/7979536f26cf9f9f3c5c622d68a3dc6b30140ae3/src/main/java/com/minecolonies/coremod/colony/Colony.java#L216): Assign this magic number 5 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L341](https://github.com/Minecolonies/minecolonies/blob/7979536f26cf9f9f3c5c622d68a3dc6b30140ae3/src/main/java/com/minecolonies/coremod/colony/Colony.java#L341): Remove this useless assignment to local variable "achievementKey". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1854)
1. ![MAJOR][MAJOR] [Colony.java#L500](https://github.com/Minecolonies/minecolonies/blob/7979536f26cf9f9f3c5c622d68a3dc6b30140ae3/src/main/java/com/minecolonies/coremod/colony/Colony.java#L500): Remove this useless assignment to local variable "achievementCompound". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1854)
1. ![MAJOR][MAJOR] [Colony.java#L621](https://github.com/Minecolonies/minecolonies/blob/7979536f26cf9f9f3c5c622d68a3dc6b30140ae3/src/main/java/com/minecolonies/coremod/colony/Colony.java#L621): Reduce the number of returns of this method 11, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [Colony.java#L1070](https://github.com/Minecolonies/minecolonies/blob/7979536f26cf9f9f3c5c622d68a3dc6b30140ae3/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1070): The Cyclomatic Complexity of this method "onWorldTick" is 17 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [Colony.java#L1106](https://github.com/Minecolonies/minecolonies/blob/7979536f26cf9f9f3c5c622d68a3dc6b30140ae3/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1106): Assign this magic number 20 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1107](https://github.com/Minecolonies/minecolonies/blob/7979536f26cf9f9f3c5c622d68a3dc6b30140ae3/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1107): Assign this magic number 60 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by marchermans — APPROVED** [review]

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 68 issues
* ![MAJOR][MAJOR] 24 major
* ![MINOR][MINOR] 41 minor
* ![INFO][INFO] 3 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/b617ccb27828d1c32311f9e5d0e5acbabf969221/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.289 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L51](https://github.com/Minecolonies/minecolonies/blob/b617ccb27828d1c32311f9e5d0e5acbabf969221/src/main/java/com/minecolonies/coremod/colony/Colony.java#L51): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 46 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L51](https://github.com/Minecolonies/minecolonies/blob/b617ccb27828d1c32311f9e5d0e5acbabf969221/src/main/java/com/minecolonies/coremod/colony/Colony.java#L51): class "Colony" has 91 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L216](https://github.com/Minecolonies/minecolonies/blob/b617ccb27828d1c32311f9e5d0e5acbabf969221/src/main/java/com/minecolonies/coremod/colony/Colony.java#L216): Assign this magic number 5 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L341](https://github.com/Minecolonies/minecolonies/blob/b617ccb27828d1c32311f9e5d0e5acbabf969221/src/main/java/com/minecolonies/coremod/colony/Colony.java#L341): Remove this useless assignment to local variable "achievementKey". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1854)
1. ![MAJOR][MAJOR] [Colony.java#L500](https://github.com/Minecolonies/minecolonies/blob/b617ccb27828d1c32311f9e5d0e5acbabf969221/src/main/java/com/minecolonies/coremod/colony/Colony.java#L500): Remove this useless assignment to local variable "achievementCompound". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1854)
1. ![MAJOR][MAJOR] [Colony.java#L621](https://github.com/Minecolonies/minecolonies/blob/b617ccb27828d1c32311f9e5d0e5acbabf969221/src/main/java/com/minecolonies/coremod/colony/Colony.java#L621): Reduce the number of returns of this method 11, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [Colony.java#L1070](https://github.com/Minecolonies/minecolonies/blob/b617ccb27828d1c32311f9e5d0e5acbabf969221/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1070): The Cyclomatic Complexity of this method "onWorldTick" is 17 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [Colony.java#L1106](https://github.com/Minecolonies/minecolonies/blob/b617ccb27828d1c32311f9e5d0e5acbabf969221/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1106): Assign this magic number 20 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1107](https://github.com/Minecolonies/minecolonies/blob/b617ccb27828d1c32311f9e5d0e5acbabf969221/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1107): Assign this magic number 60 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by Raycoms** [comment]
@Asherslab to test it you have to rerun "setupDecompWorkspace" so he knows he is in 1.12.1 now.

**Comment by Asherslab** [comment]
I've run that about 7 times now. Always 1.12 not 1.12.1

**Comment by Raycoms** [comment]
You run it and after that you ran gradle refresh?

2017-08-31 22:09 GMT+01:00 Asherslab <notifications@github.com>:

> I've run that about 7 times now. Always 1.12 not 1.12.1
>

**Comment by Asherslab** [comment]
Mhmm, I'll try with a whole new project import after school. But yeah, nada

**Comment by Raycoms** [comment]
@Asherslab did you get to test it?

**Comment by Asherslab** [comment]
screen shot 2017-09-02 at 10 03 23 am
Nope =/ refreshed, setupDecompWorkspace, everything

**Comment by Raycoms** [comment]
did you test it? Please resolve conflicts as well.

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 68 issues
* ![MAJOR][MAJOR] 24 major
* ![MINOR][MINOR] 41 minor
* ![INFO][INFO] 3 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/1771d1add82324a2eecbfce80fa59269958d62f3/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.289 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L51](https://github.com/Minecolonies/minecolonies/blob/1771d1add82324a2eecbfce80fa59269958d62f3/src/main/java/com/minecolonies/coremod/colony/Colony.java#L51): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 46 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L51](https://github.com/Minecolonies/minecolonies/blob/1771d1add82324a2eecbfce80fa59269958d62f3/src/main/java/com/minecolonies/coremod/colony/Colony.java#L51): class "Colony" has 91 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L216](https://github.com/Minecolonies/minecolonies/blob/1771d1add82324a2eecbfce80fa59269958d62f3/src/main/java/com/minecolonies/coremod/colony/Colony.java#L216): Assign this magic number 5 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L341](https://github.com/Minecolonies/minecolonies/blob/1771d1add82324a2eecbfce80fa59269958d62f3/src/main/java/com/minecolonies/coremod/colony/Colony.java#L341): Remove this useless assignment to local variable "achievementKey". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1854)
1. ![MAJOR][MAJOR] [Colony.java#L500](https://github.com/Minecolonies/minecolonies/blob/1771d1add82324a2eecbfce80fa59269958d62f3/src/main/java/com/minecolonies/coremod/colony/Colony.java#L500): Remove this useless assignment to local variable "achievementCompound". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1854)
1. ![MAJOR][MAJOR] [Colony.java#L621](https://github.com/Minecolonies/minecolonies/blob/1771d1add82324a2eecbfce80fa59269958d62f3/src/main/java/com/minecolonies/coremod/colony/Colony.java#L621): Reduce the number of returns of this method 11, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [Colony.java#L1070](https://github.com/Minecolonies/minecolonies/blob/1771d1add82324a2eecbfce80fa59269958d62f3/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1070): The Cyclomatic Complexity of this method "onWorldTick" is 17 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [Colony.java#L1106](https://github.com/Minecolonies/minecolonies/blob/1771d1add82324a2eecbfce80fa59269958d62f3/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1106): Assign this magic number 20 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1107](https://github.com/Minecolonies/minecolonies/blob/1771d1add82324a2eecbfce80fa59269958d62f3/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1107): Assign this magic number 60 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by Asherslab** [comment]
@Raycoms Ready to go!
i'll make the 1.11 from this one once it's merged (easier)

**Pull request merged**